### DISCUSSION TITLE
Are custom data types something wanted in the future?

### DISCUSSION THREAD

**Discussion by dgengtek** [nushell/nushell#9183] [General]
I wanted to ask if this is something to be considered or just not wanted at all allowing users to define their own types in the form of "structs" or "records" and could be used for type checking in custom commands or even variables.

There are already a few different types supported for custom commands https://www.nushell.sh/book/custom_commands.html#parameter-types which could basically be defined in custom records. For example the path type is described in nu as `record<parent: string, stem: string, extension: string>`.

**Comment by fdncred** [comment]
We've talked about it and thought it may be a good idea to add at some point. We'd need to discuss it on Discord in #design-discussion to arrive at a syntax if you're up for trying to implement it.

**Comment by amtoine** [comment]
i like your
```
Path = record<parent: string, stem: string, extension: string>
```
example :ok_hand:

**Reply by amtoine** [reply]
actually i'm not sure if that's planned :thinking: 
curious to see points of view from other members of the team :innocent:
